source/guides/recipes/post-build-hook.md

```diff
@@ -14,12 +14,11 @@ A more advanced implementation might pass the store paths to a user-supplied dae
 
 # Prerequisites
 
-<!-- TODO: this information will move: https://github.com/NixOS/nix/issues/7769 -->
-This tutorial assumes you have [configured an S3-compatible binary cache](https://nixos.org/manual/nix/stable/package-management/s3-substituter.html), and that the `root` user's default AWS profile can upload to the bucket.
+This tutorial assumes you have [configured an S3-compatible binary cache](https://nix.dev/manual/nix/2.22/store/types/s3-binary-cache-store#authenticated-writes-to-your-s3-compatible-binary-cache), and that the `root` user's default AWS profile can upload to the bucket.
 
 # Set up a signing key
 
-Use [`nix-store --generate-binary-cache-key`](https://nixos.org/manual/nix/stable/command-ref/nix-store/generate-binary-cache-key.html) to create a pair of cryptographic keys.
+Use [`nix-store --generate-binary-cache-key`](https://nix.dev/manual/nix/2.22/command-ref/nix-store/generate-binary-cache-key) to create a pair of cryptographic keys.
```

Comment:
I have not setup a binary cache before so I am having trouble understanding what commands are run on which machine. Is this correct?

```suggestion
Use [`nix-store --generate-binary-cache-key`](https://nix.dev/manual/nix/2.22/command-ref/nix-store/generate-binary-cache-key) to create a pair of cryptographic keys on the binary cache machine.
```

Reply:
It doesn't really matter where you create these keys, as long as you deploy the private key to the machine hosting the binary cache (which is mentioned in the next paragraph)